### PR TITLE
Converted into a pure Truffle project

### PR DESCRIPTION
**Pull request by cag**
@collinc97 Check it

### CODE REVIEW

File: migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -1,30 +1,28 @@
-var Math = artifacts.require("./Utils/Math");
-var EventFactory = artifacts.require("./Events/EventFactory.sol");
-var EtherToken = artifacts.require("./Tokens/EtherToken.sol");
-var CentralizedOracleFactory = artifacts.require("./Oracles/CentralizedOracleFactory.sol")
-var UltimateOracleFactory = artifacts.require("./Oracles/UltimateOracleFactory.sol");
-var LMSRMarketMaker = artifacts.require("./MarketMakers/LMSRMarketMaker.sol");
-var StandardMarketFactory = artifacts.require("./Markets/StandardMarketFactory.sol");
+let Math = artifacts.require('Math')
```
Comment (cag) [inline migrations/2_deploy_contracts.js]: Get it by symbol and the artifact require call won't worry about filename.

Comment (collinc97) [inline migrations/2_deploy_contracts.js]: oh cool. so this fixes all abstract migration issues?

File: test/test_markets.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        eventFactory = await EventFactory.deployed()
+        etherToken = await EtherToken.deployed()
+        standardMarketFactory = await StandardMarketFactory.deployed()
+        lmsrMarketMaker = await LMSRMarketMaker.deployed()
+    })
+
+    it('should buy and sell all outcomes', async () => {
+        // create event
+        const ipfsHash = 'QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG'
+        const oracle = utils.getParamFromTxEvent(
```
Comment (cag) [inline test/test_markets.js]: On tricky thing is getting results for things that cause a state change. Basically you have to look at events that are fired during the transaction to try and get the return value.

Comment (collinc97) [inline test/test_markets.js]: interesting. Is that a web3 api call?

File: test/test_markets.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        await etherToken.approve(market.address, funding, { from: accounts[investor] })
+
+        await market.fund(funding, { from: accounts[investor] })
+        assert.equal(await etherToken.balanceOf(accounts[investor]), 0)
+
+        // Buy outcome tokens
+        const buyer = 1
+        const outcome = 0
+        const token_count = 1e15
+        const outcome_token_cost = await lmsrMarketMaker.calcCost(market.address, outcome, token_count)
```
Comment (cag) [inline test/test_markets.js]: All numbers that come back from contract calls are BigNumbers. I've discovered that you can use them in `assert.equal` calls directly as long as _both_ args are not `BigNumber`. You can convert them to plain JS numbers by using the `.valueOf()` method, or by implicitly casting them via _some_ arithmetic ops. For some reason, if you try to use `+` with `BigNumber` and native numbers, you will end up concatenating their decimal representations. WTF.

Anyway, those are my caveats with that type.

File: migrations/2_deploy_contracts.js (hunk 1 above)
Comment (cag) [inline migrations/2_deploy_contracts.js]: I hope! ^_^'

File: test/test_markets.js (hunk 2 above)
Comment (cag) [inline test/test_markets.js]: Yes. You've already written this actually. I just factored it out into its own function.

**Comment by maurelian** [comment]
Are the rest of the tests also going to be converted to JS?

**Comment by cag** [comment]
Yes, that was what was ultimately decided upon.

**Comment by cag** [comment]
All the tests are rewritten, save for the futarchy oracle test because it doesn't seem like setting the gas limit on TestRPC does anything, the end of the difficulty oracle test because the difficulty of blocks on TestRPC is 0, and the signed message oracle test.

@collinc97 Turns out `web3` is injected by the Truffle framework into tests. I used `@digix/tempo` to wait. Majority oracle function can be called with an array of addresses.

**Comment by collinc97** [comment]
Cool. @cag Should I integrate the overloaded math function truffle test into this project? Currently, it is still in gnosis-contracts-truffle. I can also make the test more rigorous.

**Comment by cag** [comment]
Awesome, thanks @collinc97

**Pull request merged**